packages/bitcore-lib/lib/transaction/sighashschnorr.js

```diff
@@ -0,0 +1,231 @@
+'use strict';
+
+/* jshint maxparams:5 */
+
+const Signature = require('../crypto/signature');
+const BufferWriter = require('../encoding/bufferwriter');
+const Hash = require('../crypto/hash');
+const Schnorr = require('../crypto/schnorr');
+const $ = require('../util/preconditions');
+const TaggedHash = require('../crypto/taggedhash');
+const PrivateKey = require('../privatekey');
+
+/**
+ * Returns a buffer of length 32 bytes with the hash that needs to be signed
+ * for witness v1 programs as defined by:
+ * https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki
+ *
+ * @name Signing.sighash
+ * @param {Transaction} transaction the transaction to sign
+ * @param {Number} sighashType the type of the hash
+ * @param {Number} inputNumber the input index for the signature
+ * @param {Number} sigversion Taproot or Tapscript version number
+ * @param {Object} execdata Object with directives and data for creating the signature hash
```

Comment:
`@param {Record<string, any>} execdata...` gives a little better type help.

Reply:
`Record` is a typescript primitive whereas this is a vanilla JS project.